### PR TITLE
Introduce aarch64 packaging

### PR DESCRIPTION
**Pull request by jasontedor**
This commit introduces aarch64 packaging, including bundling an aarch64 JDK distribution. We had to make some interesting choices here:
 - ML binaries are not compiled for aarch64, so for now we disable ML on aarch64
 - depending on underlying page sizes, we have to disable class data sharing

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java [hunk 1]
```diff
@@ -65,8 +65,10 @@ private XPackSettings() {
     public static final Setting<Boolean> GRAPH_ENABLED = Setting.boolSetting("xpack.graph.enabled", true, Setting.Property.NodeScope);
 
     /** Setting for enabling or disabling machine learning. Defaults to true. */
-    public static final Setting<Boolean> MACHINE_LEARNING_ENABLED = Setting.boolSetting("xpack.ml.enabled", true,
-            Setting.Property.NodeScope);
+    public static final Setting<Boolean> MACHINE_LEARNING_ENABLED = Setting.boolSetting(
```
Comment (mark-vieira) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java]: Seems if we are running on arm we want this to _always_ be `false`, not just _default_ to `false`. Wouldn't this mean folks could enable ML explicitly causing funny result?

**Review by mark-vieira — APPROVED** [review]
Boy, just shows how much boilerplate is involved here. There's probably room for improvement there but luckily we don't add new distribution types very often. LGTM 👍 

FYI, we'll need to adapt the `DistributionDownloadPlugin` to the notion of an architecture as well once we want to start testing.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java (hunk 1 above)
Comment (jasontedor) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java]: I pushed 7b81079b0e3d5b92eef7cbb0dc100d0362e48e60.

File: buildSrc/src/testKit/jdk-download/reuse/build.gradle [hunk 2]
```diff
@@ -7,5 +7,6 @@ jdks {
     vendor = fakeJdkVendor
     version = fakeJdkVersion
     platform = "linux"
+    architeecture = "x64"
```
Comment (drewr) [inline buildSrc/src/testKit/jdk-download/reuse/build.gradle]: typo

File: distribution/archives/build.gradle [hunk 3]
```diff
@@ -153,49 +153,61 @@ Closure commonTarConfig = {
...
   configure(commonTarConfig)
   archiveClassifier = 'linux-x86_64'
-  with archiveFiles(modulesFiles(false, 'linux'), 'tar', 'linux', false, true)
+  with archiveFiles(modulesFiles(false, 'linux'), 'tar', 'linux', 'x64', false, true)
+}
+
+task buildOssLinuxAarch64Tar(type: SymbolicLinkPreservingTar) {
+  configure(commonTarConfig)
+  archiveClassifier = 'linux-x86_64'
+  with archiveFiles(modulesFiles(true, 'linux'), 'tar', 'linux', 'x64', true, true)
```
Comment (drewr) [inline distribution/archives/build.gradle]: Copy/paste error? I think these should be `aarch64`.

**Comment by davidkyle** [comment]
There is also the native controller to consider. 

https://github.com/elastic/elasticsearch/blob/e036ac4ffda1987e4cc7fcaff37757235a69874d/server/src/main/java/org/elasticsearch/bootstrap/Spawner.java#L61

It looks like the process is launched then dies immediately. There will be no sign of this in the logs as no code will try to connect to the controller because `xpack.ml.enabled: false` and the process failure is not logged. 

Perhaps extend the property in `plugin.descriptor.properties` to `has.native.controller.x64` and `has.native.controller. aarch64` or simply don't don't spawn the controller on aarch64

File: buildSrc/src/testKit/jdk-download/reuse/build.gradle (hunk 2 above)
Comment (jasontedor) [reply buildSrc/src/testKit/jdk-download/reuse/build.gradle]: Good catch, thanks!

File: distribution/archives/build.gradle (hunk 3 above)
Comment (jasontedor) [reply distribution/archives/build.gradle]: I pushed 0d7b4ee3748e969dc10a8fa94fcefb23874cb3f6 directly.

Comment (jasontedor) [reply distribution/archives/build.gradle]: I pushed 80c24a0d6271efc41f80b9dc63c19e58ce524be6 directly to address this. Thanks for noticing.

File: buildSrc/src/testKit/jdk-download/reuse/build.gradle (hunk 2 above)
Comment (jasontedor) [reply buildSrc/src/testKit/jdk-download/reuse/build.gradle]: I pushed 0d7b4ee3748e969dc10a8fa94fcefb23874cb3f6 directly, thanks for noticing.

**Comment by jasontedor** [comment]
@davidkyle The spawner today has support for architectures, and skips trying to fork the native controller when it doesn't exist:

https://github.com/elastic/elasticsearch/blob/91462ee306eec09d0556056ea040b5dfe079fe5e/server/src/main/java/org/elasticsearch/plugins/Platforms.java#L50-L54

https://github.com/elastic/elasticsearch/blob/91462ee306eec09d0556056ea040b5dfe079fe5e/server/src/main/java/org/elasticsearch/plugins/Platforms.java#L88

https://github.com/elastic/elasticsearch/blob/91462ee306eec09d0556056ea040b5dfe079fe5e/server/src/main/java/org/elasticsearch/bootstrap/Spawner.java#L75-L78

This means that when run on `aarch64`, since there is not an ML native controller at `x-pack-ml/platform/linux-aarch64/bin/controller`, the spawner will not even attempt to fork a native controller.

**Comment by davidkyle** [comment]
👍 thanks for the explanation @jasontedor

File: distribution/archives/build.gradle (hunk 3 above)
Comment (drewr) [reply distribution/archives/build.gradle]: Does the `archiveClassifier` also need to be updated? I thought it would be `linux-aarch64`.

Comment (jasontedor) [reply distribution/archives/build.gradle]: Thanks again. 🤦‍♀ 

I pushed ba431a0f5f958505a0bc508bd8500a838b05fe55.

**Pull request merged**